### ISSUE TITLE
How to get the perplexity of the sequence

### ISSUE THREAD

**Issue by NeonBohdan** [marella/ctransformers#48]
This project is crasy fast
But can it be applied not only to generate text but also to score it too

Is there is the way to measure the perplexity or naturalnes of the sequence, for example of a question-answer pair

**Comment by NeonBohdan** [comment]
Here is the PR that adds perplexity test over input prompt 
https://github.com/ggerganov/ggml/pull/184

**Comment by NeonBohdan** [comment]
And here one more example 
https://github.com/ggerganov/llama.cpp/pull/270

**Comment by marella** [comment]
I haven't done it myself but it should be possible to compute it using the low-level APIs `llm.eval()` and `llm.logits`.

```py
tokens = llm.tokenize(sentence)

all_logits = []
for token in tokens:
    llm.eval([token])
    all_logits.append(list(llm.logits))

# `all_logits` can be used to compute the log probabilities of tokens.
```
